packages/ai-jsx/src/batteries/sidekick/platform/large-response-handler.tsx

```diff
@@ -0,0 +1,222 @@
+import * as AI from '../../../index.js';
+import { ConversationMessage, FunctionResponse, FunctionResponseProps } from '../../../core/conversation.js';
+
+import { RecursiveCharacterTextSplitter } from 'langchain/text_splitter';
+import { getEncoding } from 'js-tiktoken';
+import yaml from 'js-yaml';
+import _ from 'lodash';
+import { UseToolsProps } from '../../use-tools.js';
+import { cohereContext, MarkdownChunkFormatter, RerankerFormatted } from '../../../lib/cohere.js';
+import { Jsonifiable } from 'type-fest';
+
+export type LengthFunction = ((text: string) => number) | ((text: string) => Promise<number>);
+
+export interface RedactedFuncionResponseMetadata {
+  isRedacted: true;
+  chunks: string[];
+}
+
+const getEncoder = _.once(() => getEncoding('cl100k_base'));
+
+function openAITokenCount(text: string) {
+  return getEncoder().encode(text).length;
+}
+
+export async function TruncateByChars(
+  {
+    children,
+    maxLength,
+  }: {
+    children: AI.Node;
+    maxLength: number;
+  },
+  { render }: AI.ComponentContext
+) {
+  const stringified = await render(children);
+  if (stringified.length <= maxLength) {
+    return stringified;
+  }
+  return `${stringified.slice(0, maxLength - 3)}...`;
+}
+
+export interface LargeFunctionResponseProps {
+  maxLength: number;
+  failedMaxLength: number;
+  numChunks: number;
+  lengthFunction?: LengthFunction;
+}
+
+async function LargeFunctionResponseHandler(
+  {
+    children,
+    maxLength = 4000,
+    failedMaxLength = 1000,
```

Comment:
No real reason. I just thought that if there is an error, we might not want to pollute the context window with it. This might not be an important concern since the effects won't be linear due to `Shrinkable`.

Reply:
I'm resolving this to merge the PR, but if you have concerns let's address them in follow up PRs.